src/hooks/connectWeb3/useWeb3React.tsx

```diff
@@ -7,7 +7,7 @@ import { Connector, Provider as Eip1193Provider, Web3ReactStore } from '@web3-re
 import { WalletConnect } from '@web3-react/walletconnect'
 import { SupportedChainId } from 'constants/chains'
 import { atom, useAtom } from 'jotai'
-import { PropsWithChildren, useEffect, useMemo } from 'react'
+import { PropsWithChildren, useEffect, useMemo, useRef } from 'react'
 import JsonRpcConnector from 'utils/JsonRpcConnector'
 
 export type Web3Connection = [Connector, Web3ReactHooks]
```

Comment:
`connections` is potentially invalid if it changes over time. Does this export need to be reconsidered?

Reply:
so, while i'm still not thrilled about this export, i checked and all the other components that use `connections` are children of `Web3ReactProvider`, and since the whole point of this pr is ensuring that this component re-renders whenever `connections` changes, there shouldn't be any immediate issues

my vote would be to merge this and consider a refactor in a later pr? can make an issue if there's agreement

Reply:
If this is the case, could there just be a hook to expose connections within the React lifecycle? I'm fine with doing this as a follow up, but it seems brittle and not future-proof, so I may just do it today.

Reply:
tracking in https://github.com/Uniswap/widgets/issues/145